### PR TITLE
fix(service-portal-health): DoD fixes

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Fixes after DoD review
* a11y fixes after aXe and SiteImprove review, still stuff to work on tho. 

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added dynamic `aria-label` switching in the `UserLanguageSwitcher` component based on selected language.
  - Introduced `linkLabel` prop in `InstitutionPanel` for better accessibility.
  - Added new messages for navigation and information in the service portal.

- **Improvements**
  - Updated `Header` component with enhanced authentication and feature flag handling.
  - Enhanced `MedicineCalculator` with new logic for disabling the "add" button.

- **Accessibility**
  - Added `role="menu"` attribute to `NavigationTree` component for better accessibility.

- **Removals**
  - Removed various feature flag-related imports and decorators across multiple services and resolvers.
  - Eliminated logging dependencies from several service classes for simplification.

### CODE REVIEW

File: libs/api/domains/rights-portal/src/lib/aidOrNutrition/aidOrNutrition.service.ts [hunk 1]
```diff
@@ -13,11 +11,7 @@ import { handle404 } from '@island.is/clients/middlewares'
 
 @Injectable()
 export class AidOrNutritionService {
-  constructor(
-    private api: AidsandnutritionApi,
-    @Inject(LOGGER_PROVIDER)
-    private logger: Logger,
-  ) {}
+  constructor(private api: AidsandnutritionApi) {}
```
Comment (disaerna) [inline libs/api/domains/rights-portal/src/lib/aidOrNutrition/aidOrNutrition.service.ts:14]: ... casing 😬

**Review by disaerna — APPROVED** [review]
Looks good, just curious about why the logging is being removed, do we have logging elsewhere?

File: libs/api/domains/rights-portal/src/lib/aidOrNutrition/aidOrNutrition.service.ts (hunk 1 above)
Comment (thorkellmani) [reply libs/api/domains/rights-portal/src/lib/aidOrNutrition/aidOrNutrition.service.ts:14]: It's from the organization swagger 🙈

**Comment by thorkellmani** [comment]
> Looks good, just curious about why the logging is being removed, do we have logging elsewhere?

We aren't logging anything in these services, and we normally just enhancedFetch log the client errors unless there's something extra we want to specifically log

**Review by saevarma — APPROVED** [review]
🚀

File: apps/service-portal/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -96,21 +99,18 @@ export const Header = ({ position }: Props) => {
...
-                              unfocusable
+                              colorScheme="white"
+                              size="small"
+                              type="button"
+                              variant="utility"
                             />
-
-                            <span className={helperStyles.srOnly}>
-                              {formatMessage(m.documents)}
-                            </span>
```
Comment (thordurhhh) [inline apps/service-portal/src/components/Header/Header.tsx:112]: Why remove this?

File: apps/service-portal/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -96,21 +99,18 @@ export const Header = ({ position }: Props) => {
...
+                            href={DocumentsPaths.ElectronicDocumentsRoot}
+                          >
                             <Button
-                              variant="utility"
-                              colorScheme="white"
-                              size="small"
                               icon="mail"
                               iconType="outline"
-                              type="span"
-                              unfocusable
```
Comment (thordurhhh) [inline apps/service-portal/src/components/Header/Header.tsx:107]: You'll get a double focus here with the button as type button focusable within the link resolver.

File: apps/service-portal/src/components/Header/Header.tsx (hunk 2 above)
Comment (thorkellmani) [reply apps/service-portal/src/components/Header/Header.tsx:112]: Mistake

File: apps/service-portal/src/components/Header/Header.tsx (hunk 3 above)
Comment (thorkellmani) [reply apps/service-portal/src/components/Header/Header.tsx:107]: I changed the `LinkResolver `to a `Link`

**Pull request merged**